src/pathfinding_service/api/rest.py

```diff
@@ -120,6 +147,102 @@ def post(self, token_network_address: str):
         return {'result': paths}, 200
 
 
+def process_payment(iou_dict, pathfinding_service):
+    if pathfinding_service.service_fee == 0:
+        return
+    if iou_dict is None:
+        raise exceptions.MissingIOU
+
+    # Basic IOU validity checks
+    iou, errors = IOU.Schema().load(iou_dict)
+    if errors:
+        raise exceptions.InvalidRequest(**errors)
+    if iou.receiver != pathfinding_service.address:
+        raise exceptions.WrongIOURecipient(expected=pathfinding_service.address)
+    if not iou.is_signature_valid():
+        raise exceptions.InvalidSignature
+
+    # Compare with known IOU
+    active_iou = pathfinding_service.database.get_iou(
+        sender=iou.sender,
+        claimed=False,
+    )
+    if active_iou:
+        if active_iou.expiration_block != iou.expiration_block:
+            raise exceptions.UseThisIOU(iou=active_iou)
+
+        expected_amount = active_iou.amount + pathfinding_service.service_fee
+    else:
+        if pathfinding_service.database.get_iou(
+            sender=iou.sender,
+            expiration_block=iou.expiration_block,
+            claimed=True,
+        ):
+            raise exceptions.IOUAlreadyClaimed
+
+        min_expiry = pathfinding_service.web3.eth.blockNumber + MIN_IOU_EXPIRY
+        if iou.expiration_block < min_expiry:
+            raise exceptions.IOUExpiredTooEarly(min_expiry=min_expiry)
+        expected_amount = pathfinding_service.service_fee
+    if iou.amount < expected_amount:
+        raise exceptions.InsufficientServicePayment(expected_amount=expected_amount)
+
+    # TODO: deposit large enough?
```

Comment:
Can you create a follow-up issue for this?

Reply:
https://github.com/raiden-network/raiden-services/issues/177